okio-assetfilesystem/src/main/kotlin/okio/assetfilesystem/AssetFileSystem.kt

```diff
@@ -154,3 +157,84 @@ private class AssetFileSystem(
     val ROOT = "/".toPath()
   }
 }
+
+private class AssetFileHandle(
+  private val assets: AssetManager,
+  private val pathString: String,
+  private var inputStream: InputStream,
+) : FileHandle(false) {
+  private var currentOffset = 0
+  private var size = -1
+
+  override fun protectedRead(
+    fileOffset: Long,
+    array: ByteArray,
+    arrayOffset: Int,
+    byteCount: Int,
+  ): Int {
+    // If we need to jump backwards or have reached the end of the file,
+    // close the existing stream and open a new one.
+    if (currentOffset > fileOffset || currentOffset == size) {
+      inputStream.close()
+      inputStream = assets.open(pathString)
+      currentOffset = 0
+    }
+
+    while (true) {
+      val skip = fileOffset - currentOffset
+      if (skip == 0L) break
+      val skipped = inputStream.skip(skip).toInt()
+      if (skipped == 0) {
+        // Since we know skip is never negative, a skip of 0 means EOF.
+        // Record this as the file size to trigger stream recreation.
+        size = currentOffset
+        throw IllegalArgumentException("fileOffset $fileOffset > size $size")
+      }
+      currentOffset += skipped
+    }
+
+    val read = inputStream.read(array, arrayOffset, byteCount)
+    if (read == -1) {
+      // A read of -1 means EOF. Record this as the file size to trigger stream recreation.
+      size = currentOffset
+    } else {
+      currentOffset += read
+    }
+    return read
+  }
+
+  override fun protectedSize(): Long {
+    if (size == -1) {
+      while (true) {
+        val skipped = inputStream.skip(128).toInt()
```

Comment:
```suggestion
        val skipped = inputStream.skip(1024 * 1024).toInt()
```

We’re not here to fuck spiders.

Reply:
Hmm I don't remember that quote from Moby Dick